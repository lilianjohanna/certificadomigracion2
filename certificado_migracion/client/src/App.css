.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}
.alert .card-content span {
	font-weight: bold;
	font-size: 1.1em;
	margin-right:4px;
}
.alert .close {
	float:right;
	position: relative;
	padding-right: 4px;
}

body.swal2-shown > [aria-hidden="true"] {
  transition: 0.1s filter;
  filter: blur(10px);
}
 body{
  padding-right: 0px !important;
 }
 .nav-fixed-scroll{
  position: fixed;
  width: 100%;
  top: 0;
  left: 0;
  z-index: 1000;
  height: 64px;
 }
 .nav-fixed-scroll nav{
   height: 64px;
   transition-property: font-size;
   transition-duration: 4s;
   transition-delay: 2s;
   font-size: 14px;
   -webkit-box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14),0 3px 1px -2px rgba(0,0,0,0.12),0 1px 5px 0 rgba(0,0,0,0.2);
   box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14),0 3px 1px -2px rgba(0,0,0,0.12),0 1px 5px 0 rgba(0,0,0,0.2);
 }
 .nav-fixed-scroll nav a{
  height: 64px;
}
 .nav-fixed-scroll nav div .brand-logo img{
  height: 50px;
}
.brand-logo img{
  height:70px;
  margin:9px 0
}
.nav-fixed-scroll .nav-tabs{
  height: 64px;
}
.nav-fixed-scroll .nav-tabs li a{
  padding-top: 0;
}

