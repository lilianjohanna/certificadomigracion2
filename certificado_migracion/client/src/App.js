import React, { Component } from "react";
import StorageCADContract from "./contracts/StorageCAD.json";
import getWeb3 from "./getWeb3";

// Librería para generar el hash sha256
import sha256 from "js-sha256";

// Librería para generar las alertas
import Swal from 'sweetalert2';
import 'animate.css';

import "./App.css";

import initial from './img/data-security.svg';
import valid from './img/valid.svg';
import invalid from './img/invalid.svg';

class App extends Component {

  state = { storageDNI: "", storageHashCAD: "", validate: null, showValidate: true, showRegister: false, numberOfRegistrations: 0, web3: null, accounts: null, contract: null, isDeploymentOwner: null, balance: null, addressContract: null };
  constructor(props) {
    super(props);
    // create a ref to store the textInput DOM element
    this.fileInput = React.createRef();
    this.textFileInput = React.createRef();
    this.textLabelDNI = React.createRef();
    this.cardValidate = React.createRef();
  }
  componentDidMount = async () => {
    Swal.fire({
      title: 'Cargando...',
      html: '<b>Inicia sesión </b> en la wallet de MetaMask<br />',
      allowEscapeKey: false,
      didOpen: () => {
        Swal.showLoading()
      },
      backdrop:true,
      allowOutsideClick: () => {
        const popup = Swal.getPopup()
        popup.classList.remove('swal2-show')
        setTimeout(() => {
          popup.classList.add('animate__animated', 'animate__headShake')
        })
        setTimeout(() => {
          popup.classList.remove('animate__animated', 'animate__headShake')
        }, 500)
        return false
      }
    });
    try {

      // Obtener el proveedor de red y la instancia web3.
      const web3 = await getWeb3();

      // Usar web3 para obtener las cuentas de los usuarios.
      const accounts = await web3.eth.getAccounts();

      // Obtener la instancia del contrato StorageCAD.
      const networkId = await web3.eth.net.getId();
      const deployedNetwork = StorageCADContract.networks[networkId];
      const instanceContract = new web3.eth.Contract(
        StorageCADContract.abi,
        deployedNetwork && deployedNetwork.address,
      );

      // Obtener la dirección del contrato
      const addressContract = deployedNetwork.address.toString();

      // Obtener el saldo de la cuenta con la que se generará las transacciones
      const balance = await web3.eth.getBalance(accounts[0]);
      const balanceETHER = web3.utils.fromWei(balance, 'ether');

      // Consultar el dueño del despliegue de los contratos y comparar si es la cuenta con la que se encuentra conectado.
      const deploymentOwner = await instanceContract.methods.owner().call();
      if (accounts[0] === deploymentOwner) {
        this.setState({ isDeploymentOwner: true });
      } else {
        this.setState({ isDeploymentOwner: false });
      };

      // Establece el estado de web3, cuentas y contrato, y luego procede
      // con la interacción del método del contrato para conocer el número de registros
      this.setState({ web3, accounts, contract: instanceContract, balance: balanceETHER, addressContract: addressContract }, this.run);
    } catch (error) {
      // Captura de errores para cualquiera de las operaciones anteriores.
      Swal.fire({
        icon: 'error',
        title: '¡Atención!',
        html: 'No se ha podido cargar',
        timer: 3000,
        showConfirmButton: false
      });
      console.error(error);
    }
  };

  // La función run usa el método numbersCADs para conocer el número de registros de CADs
  run = async () => {
    Swal.fire({
      icon: 'success',
      title: '¡Correcto!',
      timer: 700,
      showConfirmButton: false,
    });
    const { contract, isDeploymentOwner, balance } = this.state;
    if (isDeploymentOwner === true) {
      if (balance > 0) {
        Swal.fire({
          icon: 'info',
          title: 'Saldo de tu cuenta:',
          text: balance + ' ETH',
          toast: true,
          position: 'top-end',
          showConfirmButton: false,
          timer: 5000,
          timerProgressBar: true,
          didOpen: (toast) => {
            toast.addEventListener('mouseenter', Swal.stopTimer)
            toast.addEventListener('mouseleave', Swal.resumeTimer)
          }
        });
      } else {
        Swal.fire({
          icon: 'warning',
          title: 'El saldo de tu cuenta es de: ' + balance + ' ETH',
          text: 'No tienes fondos suficientes para registrar pero si puedes validar',
          toast: true,
          position: 'top-end',
          showConfirmButton: false,
          showCloseButton: true,
        });
      }
    };
    const number = await contract.methods.numbersCADs().call();
    this.setState({ numberOfRegistrations: number });
  };

  // La función handleSubmit  usa los métodos para crear el registro de un CAD en la red Blockchain
  // teniendo en cuenta si ya se encuentra registrado o no
  handleSubmit = async () => {
    const { storageDNI, storageHashCAD, accounts, contract } = this.state;
    const registered = await contract.methods.findHash(storageHashCAD).call();

    // Se comprueba que los campos no esten vacios
    if (storageDNI === "" || storageHashCAD === "") {
      Swal.fire({
        icon: 'warning',
        title: '¡Atención!',
        text: 'Los campos deben estar completos para registrar el certificado académico digital en la red Blockchain',
        confirmButtonText: 'Aceptar',
      });
    }
    // Se comprueba que no este registrado
    else if (registered === false) {
      Swal.fire({
        title: 'Espere...',
        html: '<b>Confirmar</b> la transacción y esperar el mensaje de confirmación (esto tomará un tiempo)... <br />',
        didOpen: () => {
          Swal.showLoading()
        },
        backdrop:true,
        allowOutsideClick: () => {
          const popup = Swal.getPopup()
          popup.classList.remove('swal2-show')
          setTimeout(() => {
            popup.classList.add('animate__animated', 'animate__headShake')
          })
          setTimeout(() => {
            popup.classList.remove('animate__animated', 'animate__headShake')
          }, 500)
          return false
        }
      });
      try {
        let result = await contract.methods.registerCAD(storageDNI, storageHashCAD, accounts[0]).send({ from: accounts[0] });
        Swal.fire({
          icon: 'success',
          title: '¡Correcto!',
          text: 'El certificado académico digital se registró en la red Blockchain con éxito',
          confirmButtonText: 'Aceptar',
        });
        this.resetForm();
        console.log(result);
      } catch (error) {
        Swal.fire({
          icon: 'warning',
          title: '¡Atención!',
          html: 'Error al registrar el certificado académico digital <br />' +
            'Inténtelo más tarde o revise el siguiente mensaje:' +
            '<hr style="border:1px dashed #dfdfdf; width:90%"></hr>' +
            '<span style="color:red">Error: </span>' +
            error.message,
          confirmButtonText: 'Aceptar'
        });
      };

    } else {
      Swal.fire({
        icon: 'warning',
        title: '¡Atención!',
        text: 'El certificado académico digital ya se encuentra registrado en la red Blockchain',
        confirmButtonText: 'Aceptar',
      });
    }

    // Se actualiza el estado del número de registros para mostrar en pantalla
    const number = await contract.methods.numbersCADs().call();
    this.setState({ numberOfRegistrations: number });
  };

  // Función para obtener los valores de los inputs en el formulario
  handleInputChange = (event) => {
    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;
    this.setState({
      [name]: value
    });
  };

  // Función para validar si el CAD es o no autentico
  handleValidate = async () => {
    const { storageDNI, storageHashCAD, contract } = this.state;
    if (storageDNI === "" || storageHashCAD === "") {
      Swal.fire({
        icon: 'warning',
        title: '¡Atención!',
        text: 'Los campos deben estar completos para validar la autenticidad del certificado académico digital',
        confirmButtonText: 'Aceptar',
      });
    } else {
      try {
        const response = await contract.methods.validateCAD(storageDNI, storageHashCAD).call();
        const Toast = Swal.mixin({
          toast: true,
          position: 'top-end',
          showConfirmButton: false,
          timer: 5000,
          timerProgressBar: true,
          didOpen: (toast) => {
            toast.addEventListener('mouseenter', Swal.stopTimer)
            toast.addEventListener('mouseleave', Swal.resumeTimer)
          }
        });
        Toast.fire({
          icon: 'info',
          title: 'Validación de autenticidad realizada, observe el resultado en pantalla'
        });
        this.setState({ validate: response });
        const card = this.cardValidate.current;
        card.classList.add('animate__animated', 'animate__backInRight');
        setTimeout(() => card.classList.remove('animate__animated', 'animate__backInRight'), 1000);
      } catch (error) {
        Swal.fire({
          icon: 'warning',
          title: '¡Atención!',
          html: 'Error al validar el certificado académico digital <br />' +
            'Inténtelo más tarde o revise el siguiente mensaje:' +
            '<hr style="border:1px dashed #dfdfdf; width:90%"></hr>' +
            '<span style="color:red">Error: </span>' +
            error.message,
          confirmButtonText: 'Aceptar'
        });
      };
    }
  };

  // Función para mostrar el formulario de registro
  handleChangeValidate = async () => {
    this.setState({ showValidate: true });
    this.setState({ showRegister: false });
    this.setState({ storageDNI: "" });
    this.setState({ storageHashCAD: "" });
    this.setState({ validate: null });
  };

  // Función para mostrar el formulario de validación
  handleChangeRegister = async () => {
    this.setState({ showValidate: false });
    this.setState({ showRegister: true });
    this.setState({ storageDNI: "" });
    this.setState({ storageHashCAD: "" });
    this.setState({ validate: null });
  };
  // Método para leer el documento en formato PDF
  readFile = (e) => {
    const input = this.fileInput.current;
    const inputTextFile = this.textFileInput.current;
    const extValidate = /(.pdf)$/i;
    if (!extValidate.exec(input.value)) {
      inputTextFile.classList.remove('invalid');
      inputTextFile.classList.add('invalid');
      this.setState({ storageHashCAD: "" });
    } else {
      inputTextFile.classList.remove('invalid');
      inputTextFile.classList.add('valid');
      const file = e.target.files[0];
      if (!file) return;

      const fileReader = new FileReader();

      // El formato al que se convierte el terxto es a un String binario
      fileReader.readAsBinaryString(file);

      fileReader.onload = () => {
        // El algoritmo de función hash criptográfica seleccionado es el SHA256 o HASH256
        const hash256File = sha256(fileReader.result);
        this.setState({ storageHashCAD: hash256File });
      }

      fileReader.onerror = () => {
        Swal.fire({
          title: 'Error',
          icon: 'error',
          text: fileReader.error,
          allowEscapeKey: false,
          confirmButtonText: 'Aceptar',
          backdrop:true,
          allowOutsideClick: () => {
            const popup = Swal.getPopup()
            popup.classList.remove('swal2-show')
            setTimeout(() => {
              popup.classList.add('animate__animated', 'animate__headShake')
            })
            setTimeout(() => {
              popup.classList.remove('animate__animated', 'animate__headShake')
            }, 500)
            return false
          }
        }).then((result) => {
          if (result.isConfirmed) {
            window.location.reload(true);
          }
        });
        console.log(fileReader.error);
      }
    }
  };
  resetForm = () => {
    this.myFormRef.reset();
    this.setState({ storageDNI: "" });
    this.setState({ storageHashCAD: "" });
    this.textLabelDNI.current.classList.add('active');
    this.setState({ validate: null });
  };

  render() {
    let urlAddressContract = "https://rinkeby.etherscan.io/address/" + this.state.addressContract;

    // Se comprueba la conexión con web3
    if (!this.state.web3) {
      return (
        <div className="container row">
          <div className="col m2 hide-on-small-and-down"></div>
          <div className="col s12 m8">
            <div className="alert card amber lighten-4 brown-text hide-on-small-and-down">
              <div className="close">
                <i data-dismiss="alert" aria-label="Close" className="fas fa-times"></i>
              </div>
              <div className="card-content">
                <p>
                  <i className="fas fa-exclamation-circle"></i> <span>Atención:</span>
                  Compruebe que su navegador sea <strong>Chrome</strong> o <strong>Firefox</strong>.
                </p>
              </div>
            </div>
            <div className="alert card amber lighten-4 brown-text hide-on-small-and-down">
              <div className="close">
                <i data-dismiss="alert" aria-label="Close" className="fas fa-times"></i>
              </div>
              <div className="card-content">
                <p>
                  <i className="fas fa-exclamation-circle"></i> <span>Atención:</span>
                  Tener instalado la extensión <strong>MetaMask</strong> (<strong>iniciar sesión</strong> o <strong>crear cuenta</strong>), más información <a href="https://metamask.io/" target="_blank" rel="noopener noreferrer">aquí</a>.
                </p>
              </div>
            </div>
            <div className="alert card amber lighten-4 brown-text show-on-small show-on-medium hide-on-med-and-up">
              <div className="close">
                <i data-dismiss="alert" aria-label="Close" className="fas fa-times"></i>
              </div>
              <div className="card-content">
                <p>
                  <i className="fas fa-exclamation-circle"></i> <span>Atención:</span>
                  Si está intentando ingresar desde el navegador de su móvil, hágalo por medio de la aplicación móvil de MetaMask, más información <a href="https://metamask.io/" target="_blank" rel="noopener noreferrer">aquí</a>.
                </p>
              </div>
            </div>
            <div className="alert card amber lighten-4 brown-text">
              <div className="close">
                <i data-dismiss="alert" aria-label="Close" className="fas fa-times"></i>
              </div>
              <div className="card-content">
                <p>
                  <i className="fas fa-exclamation-circle"></i> <span>Atención:</span>Estar conectado a la red <strong>Rinkeby</strong>
                </p>
              </div>
            </div>
          </div>
          <div className="col m2 hide-on-small-and-down"></div>
        </div>
      )
    }
    return (
      <div className="container">
        {/* Sección de título */}
        <div className="center">
          <h5 className="principal-title">Validación de autenticidad de los certificados académicos digitales</h5>
          <br />
        </div>
        <div className="row">
          {this.state.showRegister === true && this.state.isDeploymentOwner === true &&
            <div className="row">
              <div className="col s12 m6">
                <ul className="tabs container-one">
                  <li className="tab col s6"><button onClick={this.handleChangeValidate}>Validar certificados</button></li>
                  <li className="tab col s6 active"><button className="active" onClick={this.handleChangeRegister}>Registrar certificados</button></li>
                </ul>
              </div>
            </div>
          }
          {this.state.showValidate === true && this.state.isDeploymentOwner === true &&
            <div className="row">
              <div className="col s12 m6">
                <ul className="tabs container-one">
                  <li className="tab col s6 active"><button data-testid="btnChangeValidate" className="active" onClick={this.handleChangeValidate}>Validar certificados</button></li>
                  <li className="tab col s6"><button data-testid="btnChangeRegister" onClick={this.handleChangeRegister}>Registrar certificados</button></li>
                </ul>
              </div>
            </div>
          }
          <div className="row">
            <div className="col s12 m6">
              <span>Complete los campos correctamente:</span>
            </div>
          </div>

          <div className="row container-two">
            {/* Sección de registro de CAD */}
            {this.state.showRegister === true && this.state.isDeploymentOwner === true &&
              <div className="col s12 m6">
                <form ref={(el) => this.myFormRef = el}>
                  <div className="row">
                    <div className="input-field">
                      <input id="dni" type="text" name="storageDNI" value={this.state.storageDNI} onChange={this.handleInputChange} />
                      <label className="active" ref={this.textLabelDNI}>Número de DNI:</label>
                      <span className="helper-text">Ingrese el número de DNI</span>
                    </div>
                    <div className="file-field input-field">
                      <div className="btn">
                        <span>Subir archivo</span>
                        <input type="file" name="inputfile" ref={this.fileInput} multiple={false} onChange={this.readFile} accept=".pdf" />
                      </div>
                      <div className="file-path-wrapper">
                        <input className="file-path" ref={this.textFileInput} placeholder="No se ha elegido ningún archivo" type="text" />
                        <span className="helper-text" data-error="El certificado seleccionado debe ser un archivo PDF" data-success="El certificado seleccionado tiene el formato adecuado" >Subir el certificado académico digital en formato PDF</span>
                      </div>
                    </div>
                    {this.state.storageHashCAD !== "" &&
                      <div className="input-field">
                        <span className="helper-text"><i className="fas fa-info-circle"></i> SHA-256 del certificado: {this.state.storageHashCAD}</span>
                      </div>
                    }
                  </div>
                  <div className="center">
                    <button type="button" data-test-id="test-button-register" className="btn btnBlueUNL" disabled={this.state.storageDNI === "" || this.state.storageHashCAD === ""} onClick={this.handleSubmit}>Registrar en Blockchain</button>
                  </div>
                </form>
                <div className="alert card blue lighten-4 blue-text text-darken-3">
                  <div className="card-content">
                    <p><i className="fas fa-info-circle"></i> <span>Información:</span> Actualmente existen <span>{this.state.numberOfRegistrations}</span>certificados académicos digitales registrados en la red Rinkeby de Ethereum.</p>
                  </div>
                </div>
              </div>
            }

            {/* Sección de validación de CAD */}
            {this.state.showValidate === true &&
              <div className="col s12 m6">
                <form ref={(el) => this.myFormRef = el}>
                  <div className="row">
                    <div className="input-field">
                      <input id="dni" type="text" name="storageDNI" value={this.state.storageDNI} onChange={this.handleInputChange} />
                      <label className="active" ref={this.textLabelDNI}>Número de DNI:</label>
                      <span className="helper-text">Ingrese el número de DNI</span>
                    </div>
                    <div className="file-field input-field">
                      <div className="btn">
                        <span>Subir archivo</span>
                        <input type="file" name="inputfile" ref={this.fileInput} multiple={false} onChange={this.readFile} accept=".pdf" />
                      </div>
                      <div className="file-path-wrapper">
                        <input className="file-path" ref={this.textFileInput} placeholder="No se ha elegido ningún archivo" type="text" />
                        <span className="helper-text" data-error="El certificado seleccionado debe ser un archivo PDF" data-success="El certificado seleccionado tiene el formato adecuado" >Subir el certificado académico digital en formato PDF</span>
                      </div>
                    </div>
                  </div>
                  <div className="center">
                    <button type="button" data-test-id="test-button-validate" className="btn btnBlueUNL" disabled={this.state.storageDNI === "" || this.state.storageHashCAD === ""} onClick={this.handleValidate}>Validar</button>
                  </div>
                </form>
              </div>
            }
            <div className="col s12 m1 hide-on-small-and-down"></div>
            <div className="col s12 m1 verticalLine hide-on-small-and-down"></div>

            {/* Sección de inicio indicando que el sistema está listo */}
            {this.state.validate === null &&
              <div className="col s12 m4 center">
                <div className="card-content">
                  <p>
                    <img className='img-initial' src={initial} alt="El sistema está listo" /> <br />
                    <span className="second-title-cost">El sistema está listo</span>
                  </p>
                </div>
                {this.state.showRegister === true && this.state.isDeploymentOwner === true &&
                  <div className="card-action">
                    <a href={urlAddressContract} target="_blank" rel="noopener noreferrer">Visualizar las transacciones en Etherscan <i className="fas fa-external-link-alt"></i> </a>
                  </div>
                }
              </div>
            }

            {/* Sección de validación positiva de CAD */}
            {this.state.validate === true &&
              <div className="col s12 m4 center">
                <div className="row " >
                  <div className="col s12">
                    <div className="card" ref={this.cardValidate}>
                      <div className="card-content">
                        <span className="">El certificado académico digital:</span> <br /> <br />
                        <img className='img-result' src={valid} alt="Documento auténtico" /> <br />
                        <span className="second-title-cost">Es auténtico</span>
                      </div>
                      <div className="card-action">
                        <button className="btn" onClick={this.resetForm}>Validar otro documento</button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            }
            {/* Sección de validación negativa de CAD */}
            {this.state.validate === false &&
              <div className="col s12 m4 center">
                <div className="row " >
                  <div className="col s12">
                    <div className="card" ref={this.cardValidate}>
                      <div className="card-content">
                        <span className="">El certificado académico digital:</span> <br /> <br />
                        <img className='img-result' src={invalid} alt="Documento no auténctico" /> <br />
                        <span className="red-text-UNL">No es auténtico</span>
                      </div>
                      <div className="card-action">
                        <button className="btn" onClick={this.resetForm}>Validar otro documento</button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            }
          </div>
        </div>
      </div >
    );
  }
}

export default App;
