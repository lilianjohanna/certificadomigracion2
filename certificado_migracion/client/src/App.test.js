import React from 'react';
import App from './App';
import '@testing-library/jest-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('Pruebas unitarias a: <App />', () => {
  let stateConecction = { web3: true };
  let wrapperNoConnection = shallow(<App />);
  let wrapperConnection = shallow(<App />).setState(stateConecction);
  let stateRegisterCDA;
  let stateValidateCDA;
  let stateShowValidate;
  let stateShowRegister;
  let stateCDAIsAuthentic;
  let stateCDAIsNoAuthentic;

  beforeEach(() => {
    wrapperNoConnection = shallow(<App />);
    wrapperConnection = shallow(<App />).setState(stateConecction);
    stateShowValidate = { web3: true, showRegister: false, showValidate: true , isDeploymentOwner: true};
    stateShowRegister = { web3: true, showRegister: true, showValidate: false , isDeploymentOwner: true};
    stateCDAIsAuthentic = { web3: true, showRegister: false, showValidate: true, validate: true };
    stateCDAIsNoAuthentic = { web3: true, showRegister: false, showValidate: true, validate: false };
    stateRegisterCDA = {
      web3: true,
      showRegister: true,
      showValidate: false,
      storageDNI: '1104717572',
      storageHashCAD: '19607064cec2aa57d49d848fb09a720aaf511c7b01eb18577a115ba0b5d5d7bc'
    };

    stateValidateCDA = {
      web3: true,
      showRegister: false,
      showValidate: true,
      storageDNI: '1104717572',
      storageHashCAD: '19607064cec2aa57d49d848fb09a720aaf511c7b01eb18577a115ba0b5d5d7bc'
    };
  });

  test('Renderizar <App /> al no estar conectado a web3, cuentas, contrato o metamask ', () => {
    expect(wrapperNoConnection).toMatchSnapshot();
    expect(wrapperNoConnection.find(App)).toBeDefined();
  });

  test('Renderizar <App /> correctamente al estar conectado correctamente', () => {
    expect(wrapperConnection).toMatchSnapshot();
    expect(wrapperConnection.find(App)).toBeDefined();
  });

  test('Mostrar formulario para validar un CDA', () => {
    const wrapper = shallow(<App />).setState(stateShowValidate);
    const counterText = wrapper.find('button').at(2).text().trim();
    expect(counterText).toBe('Validar');
    expect(wrapper.find('button').at(0).hasClass('active')).toEqual(true);
    expect(wrapper.find('button').at(1).hasClass('active')).toEqual(false);
  });

  test('Mostrar formulario para registrar un CDA', () => {
    const wrapper = shallow(<App />).setState(stateShowRegister);
    const counterText = wrapper.find('button').at(2).text().trim();
    expect(wrapper.contains(<input className="file-path" placeholder="No se ha elegido ningún archivo" type="text"/>)).toEqual(true);
    expect(counterText).toBe('Registrar en Blockchain');
    expect(wrapper.find('button').at(0).hasClass('active')).toEqual(false);
    expect(wrapper.find('button').at(1).hasClass('active')).toEqual(true);
  });

  test('Mostrar el mensaje adecuado cuando el CDA es autentico', () => {
    let wrapper = shallow(<App />).setState(stateCDAIsAuthentic);
    const counterText = wrapper.find('span').at(5).text().trim();
    expect(counterText).toBe('Es auténtico');
  });

  test('Mostrar el mensaje adecuado cuando el CDA no es autentico', () => {
    let wrapper = shallow(<App />).setState(stateCDAIsNoAuthentic);
    const counterText = wrapper.find('span').at(5).text().trim();
    expect(counterText).toBe('No es auténtico');
  });

  test('Llamar a componentDidMount()', async () => {
    const wrapper = mount(<App />).setState(stateConecction);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'componentDidMount')
    expect(instance.componentDidMount).toHaveBeenCalledTimes(0);
  });

  test('Llamar a la función run()', () => {
    const wrapper = mount(<App />).setState(stateConecction);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'run')
    expect(instance.run).toHaveBeenCalledTimes(0);
  });

  test('Llamar a la función handleSubmit()', () => {
    const wrapper = mount(<App />).setState(stateRegisterCDA);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'handleSubmit')
    expect(instance.handleSubmit).toHaveBeenCalledTimes(0);
  });

  test('Llamar a la función handleValidate()', () => {
    const wrapper = mount(<App />).setState(stateValidateCDA);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'handleValidate')
    expect(instance.handleValidate).toHaveBeenCalledTimes(0);
  });

  test('Llamar a la función handleChangeValidate()', () => {
    const handleChangeValidate = jest.fn();
    const wrapper = shallow((<button onClick={handleChangeValidate}>Validar certificados</button>));
    wrapper.find('button').simulate('click');
    expect(handleChangeValidate.mock.calls.length).toEqual(1);
  });

  test('Llamar a la función handleChangeRegister()', () => {
    const handleChangeRegister = jest.fn();
    const wrapper = shallow((<button onClick={handleChangeRegister}>Registrar certificados</button>));
    wrapper.find('button').simulate('click');
    expect(handleChangeRegister.mock.calls.length).toEqual(1);
  });

  test('Llamar a la función readFile()', () => {
    const wrapper = mount(<App />).setState(stateValidateCDA);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'readFile')
    expect(instance.readFile).toHaveBeenCalledTimes(0);
  });
});
