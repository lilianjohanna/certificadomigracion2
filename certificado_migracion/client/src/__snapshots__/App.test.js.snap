// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias a: <App /> Renderizar <App /> al no estar conectado a web3, cuentas, contrato o metamask  1`] = `
<div
  className="container row"
>
  <div
    className="col m2 hide-on-small-and-down"
  />
  <div
    className="col s12 m8"
  >
    <div
      className="alert card amber lighten-4 brown-text hide-on-small-and-down"
    >
      <div
        className="close"
      >
        <i
          aria-label="Close"
          className="fas fa-times"
          data-dismiss="alert"
        />
      </div>
      <div
        className="card-content"
      >
        <p>
          <i
            className="fas fa-exclamation-circle"
          />
           
          <span>
            Atención:
          </span>
          Compruebe que su navegador sea 
          <strong>
            Chrome
          </strong>
           o 
          <strong>
            Firefox
          </strong>
          .
        </p>
      </div>
    </div>
    <div
      className="alert card amber lighten-4 brown-text hide-on-small-and-down"
    >
      <div
        className="close"
      >
        <i
          aria-label="Close"
          className="fas fa-times"
          data-dismiss="alert"
        />
      </div>
      <div
        className="card-content"
      >
        <p>
          <i
            className="fas fa-exclamation-circle"
          />
           
          <span>
            Atención:
          </span>
          Tener instalado la extensión 
          <strong>
            MetaMask
          </strong>
           (
          <strong>
            iniciar sesión
          </strong>
           o 
          <strong>
            crear cuenta
          </strong>
          ), más información 
          <a
            href="https://metamask.io/"
            rel="noopener noreferrer"
            target="_blank"
          >
            aquí
          </a>
          .
        </p>
      </div>
    </div>
    <div
      className="alert card amber lighten-4 brown-text show-on-small show-on-medium hide-on-med-and-up"
    >
      <div
        className="close"
      >
        <i
          aria-label="Close"
          className="fas fa-times"
          data-dismiss="alert"
        />
      </div>
      <div
        className="card-content"
      >
        <p>
          <i
            className="fas fa-exclamation-circle"
          />
           
          <span>
            Atención:
          </span>
          Si está intentando ingresar desde el navegador de su móvil, hágalo por medio de la aplicación móvil de MetaMask, más información 
          <a
            href="https://metamask.io/"
            rel="noopener noreferrer"
            target="_blank"
          >
            aquí
          </a>
          .
        </p>
      </div>
    </div>
    <div
      className="alert card amber lighten-4 brown-text"
    >
      <div
        className="close"
      >
        <i
          aria-label="Close"
          className="fas fa-times"
          data-dismiss="alert"
        />
      </div>
      <div
        className="card-content"
      >
        <p>
          <i
            className="fas fa-exclamation-circle"
          />
           
          <span>
            Atención:
          </span>
          Estar conectado a la red 
          <strong>
            Rinkeby
          </strong>
        </p>
      </div>
    </div>
  </div>
  <div
    className="col m2 hide-on-small-and-down"
  />
</div>
`;

exports[`Pruebas unitarias a: <App /> Renderizar <App /> correctamente al estar conectado correctamente 1`] = `
<div
  className="container"
>
  <div
    className="center"
  >
    <h5
      className="principal-title"
    >
      Validación de autenticidad de los certificados académicos digitales
    </h5>
    <br />
  </div>
  <div
    className="row"
  >
    <div
      className="row"
    >
      <div
        className="col s12 m6"
      >
        <span>
          Complete los campos correctamente:
        </span>
      </div>
    </div>
    <div
      className="row container-two"
    >
      <div
        className="col s12 m6"
      >
        <form>
          <div
            className="row"
          >
            <div
              className="input-field"
            >
              <input
                id="dni"
                name="storageDNI"
                onChange={[Function]}
                type="text"
                value=""
              />
              <label
                className="active"
              >
                Número de DNI:
              </label>
              <span
                className="helper-text"
              >
                Ingrese el número de DNI
              </span>
            </div>
            <div
              className="file-field input-field"
            >
              <div
                className="btn"
              >
                <span>
                  Subir archivo
                </span>
                <input
                  accept=".pdf"
                  multiple={false}
                  name="inputfile"
                  onChange={[Function]}
                  type="file"
                />
              </div>
              <div
                className="file-path-wrapper"
              >
                <input
                  className="file-path"
                  placeholder="No se ha elegido ningún archivo"
                  type="text"
                />
                <span
                  className="helper-text"
                  data-error="El certificado seleccionado debe ser un archivo PDF"
                  data-success="El certificado seleccionado tiene el formato adecuado"
                >
                  Subir el certificado académico digital en formato PDF
                </span>
              </div>
            </div>
          </div>
          <div
            className="center"
          >
            <button
              className="btn btnBlueUNL"
              data-test-id="test-button-validate"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              Validar
            </button>
          </div>
        </form>
      </div>
      <div
        className="col s12 m1 hide-on-small-and-down"
      />
      <div
        className="col s12 m1 verticalLine hide-on-small-and-down"
      />
      <div
        className="col s12 m4 center"
      >
        <div
          className="card-content"
        >
          <p>
            <img
              alt="El sistema está listo"
              className="img-initial"
              src="data-security.svg"
            />
             
            <br />
            <span
              className="second-title-cost"
            >
              El sistema está listo
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
